### PR TITLE
HLRC XPack Protocol clean up: Licence, Misc

### PR DESCRIPTION
**Pull request by vladimirdolzhenko**
Move parts related to XPack Licence and Misc from protocol to client

Relates #34451

### CODE REVIEW

File: client/rest-high-level/src/test/java/org/elasticsearch/client/LicensingIT.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        assertThat(map.containsKey("license"), equalTo(true));
+
+        @SuppressWarnings("unchecked")
+        final Map<String, Object> license = (Map<String, Object>) map.get("license");
+        assertThat(license.get("status"), equalTo("active"));
+        assertThat(license.get("type"), equalTo("trial"));
+    }
+
+    public void testPutLicense() throws Exception {
+        // TODO
```
Comment (vladimirdolzhenko) [inline client/rest-high-level/src/test/java/org/elasticsearch/client/LicensingIT.java]: we need start trial ( https://github.com/elastic/elasticsearch/pull/33406 ) as a roll back here

File: client/rest-high-level/src/main/java/org/elasticsearch/client/license/GetLicenseRequest.java [hunk 2]
```diff
@@ -16,16 +16,23 @@
...
  * under the License.
  */
-package org.elasticsearch.protocol.xpack;
+package org.elasticsearch.client.license;
 
-import org.elasticsearch.action.ActionRequestValidationException;
-import org.elasticsearch.action.support.master.MasterNodeRequest;
+import org.elasticsearch.client.Validatable;
 
-public class XPackUsageRequest extends MasterNodeRequest<XPackUsageRequest> {
```
Comment (hub-cap) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/license/GetLicenseRequest.java]: lolwut... thanks for nothing, git.

File: client/rest-high-level/src/main/java/org/elasticsearch/client/license/PutLicenseResponse.java [hunk 3]
```diff
@@ -58,7 +55,7 @@
     });
 
     static {
-        PARSER.declareBoolean(constructorArg(), new ParseField("acknowledged"));
```
Comment (hub-cap) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/license/PutLicenseResponse.java]: i actually dont mind having these defined here if we dont use them anywhere else.

**Review by hub-cap — CHANGES_REQUESTED** [review]
I think we can split this into 2 PRs, one for the refactor,  and one for the protocol-test stuff. I dont think they are related, either. <3 for the cleanup!!!

**Comment by vladimirdolzhenko** [comment]
@elasticmachine test this please

**Comment by vladimirdolzhenko** [comment]
@elasticmachine test this please

**Comment by vladimirdolzhenko** [comment]
run gradle build tests please

**Comment by vladimirdolzhenko** [comment]
run gradle build tests please

**Comment by vladimirdolzhenko** [comment]
@hub-cap I've raised another PR for cross xpack-hlrc tests (actually it's PoC) #34547 . Could you pls have another look into this ?

**Comment by vladimirdolzhenko** [comment]
run gradle build tests please

**Review by hub-cap — APPROVED** [review]
Looks good now, ty for splitting.

**Comment by vladimirdolzhenko** [comment]
thanks @hub-cap for the review

**Pull request merged**